Fix missing `docs` config files

sahilbhola14 opened the pull request:
## Purpose
-  Doc configs were missing in #18 which would potentially result in failed doc building.
-  This PR also restructures the directory structure to hierarchical.


 Closes #25 

Sphinx config files were missing in #18 CI which would potentially fail the doc building.
Before this PR, this directory structure was modular. This PR introduces hierarchical directory structure.


## Type of change

- Bugfix (non-breaking change which fixes an issue)
- Repository chores (changes made to improve repository/workflow quality)


## Checklist
_Put an `x` in the boxes that apply._

- [x] I have addressed all the issues entirely

sahilbhola14 committed:
(fix/pre-commit) Make Flake and Black 88 character

Make Flake and Black 88 character linewidth

sahilbhola14 committed:
(fix/pre-commit) Fix line-width command Flake

Fixed the Flake line-width command name

sahilbhola14 committed:
(fix/docs) Add missing doc config files

Added missing `docs` config files

dorukaks commented on docs/introduction.rst:
```diff
@@ -0,0 +1,29 @@
+**ICF-MI** combines two statistical algorithms, **TT-ICE** which is a tensor-train decomposition alogorithm and **GIM** which is a model identification alogorithm.
+The Algorithm has the following salient features:
+
+* **TT-ICE** is a tensor-train decomposition algorithm which is a generalization of the popular **ICE** algorithm. It is a non-parametric algorithm which can be used to decompose a high-dimensional function into a sum of low-dimensional functions.
```
Remind me to correct those :smile:  , ICE stands for [Incremental Core Expansion](https://arxiv.org/abs/2211.12487). Lmk if you want me do the changes before or after merging this PR.

dorukaks replied:
If you want those addressed, I'll quickly checkout the branch and do the changes in the morning

sahilbhola14 replied:
Oh, Lol. You can make the changes by checking the branch associated with this pull request, i.e. `fix/sahil/25`. Let me know when you make your commit

dorukaks replied:
Updated my part :+1:  good to go

dorukaks commented:
Let us know which one you are going to merge first #28 or #29

sahilbhola14 commented:
#28 needs to be merged first

dorukaks commented:
And @sahilbhola14 please don't hesitate to force any directory structure through this PR, we can close PR #26 without merging and reopen to comply with the discussed repository and branching structure.

Hopefully #28 and #29 will be the last PR's into main and we can start working on dev branch while creating our individual features

sahilbhola14 commented:
@dorukaks Okay, I will modify the directory structure as well in this PR

sahilbhola14 committed:
(refactor) Make directory structure hierarchichal

Refactor directory structure to hierarchichal

sahilbhola14 committed:
(refactor) Add hierarchical folder in doc building

Changed the compile statements to include hierarchical
structure for doc building

dorukaks reviewed (approved):
As far as I've seen, this looks good. Let's create dev branch after merging this and #28. Let's gooo

dorukaks committed:
fix(introduction) updated TT-ICE description

Changed the description for TT-ICE in introduction.rst file.

dorukaks commented:
@cfkawa please refer to the [guideline](https://github.com/PinguDevTeam/ICF-MI-framework/blob/main/formats/branching.md#pull-requests-prs) for further information on how to merge PRs.
